File: Problem 1 - Multiples of 3 and 5/mul_3_5.cpp
// Sum of all the multiples of 3 or 5

#include <iostream>

using namespace std;

uint64_t sumN(uint64_t n) {
	return (n * (n+1))/2;
}

uint64_t sumDivisibleBy(uint64_t num, uint64_t upto) {
	uint64_t linearUpto = (upto-1)/num;
	return (num * sumN(linearUpto));
}

int main() {
	uint64_t inp;
	cin >> inp;
	uint64_t sum = (sumDivisibleBy(3, inp) + sumDivisibleBy(5, inp)) - sumDivisibleBy(15, inp); 
	cout << sum << endl;
}

File: Problem 1 - Multiples of 3 and 5/mul_3_5.py
# Sum of num divisble by 3 and 5 upto 1000

def sumN(n):
    return ((n * (n+1)) // 2)

def sumDivisibleBy(num, upto):
    linearUpto = (upto-1)//num
    return num * sumN(linearUpto)

upto = int(input())
ans = sumDivisibleBy(3,upto) + sumDivisibleBy(5,upto) - sumDivisibleBy(15,upto)
print(ans)
